feat(atoms): add ListEventTypes atom to enable event type management (#25047)

pedroccastro opened the pull request:
## What does this PR do?

This PR adds the `ListEventTypes` atom to Cal.com Platform, enabling partners to display and manage event types within their applications.

**Key features:**
📋 Display all event types with title, description, and duration
🗑️ Delete event types with browser confirmation
✏️ Optional navigation to edit pages
📭 Empty and error states with proper feedback
✅ 11 unit tests

- Fixes #25047
- Fixes CAL-6730

## Visual Demo

> **Note:** Screenshots updated to reflect latest changes:
> - ConfirmationDialog instead of native confirm
> - Badge component for duration display
> - Improved duration formatting (e.g., "1h 15m")

### List View with Multiple Event Types
![List View](https://github.com/user-attachments/assets/e6cfc5b8-09b7-407e-bcd7-fec0026f7476)
*Shows event types list with title, description, and duration*

### Dropdown Menu Options
![Dropdown Menu](https://github.com/user-attachments/assets/f7799eeb-ba19-4ff3-b3aa-a40473a6e7ed)
*Edit and Delete options with proper styling*

### Delete Confirmation
![Delete Dialog](https://github.com/user-attachments/assets/eeb66ef6-bbac-4956-8b98-072ac5ef86f0)
*Browser confirmation dialog before deletion*

## Technical Changes

### Backend (API)
**File:** `packages/trpc/server/routers/viewer/eventTypes/bulkEventFetch.handler.ts`

Added fields to event types response:
```diff
{
  id: 1,
  title: "30min",
+ slug: "30min",
+ description: "Quick meeting",
+ length: 30,
  locations: null,
  logo: "/app-store/dailyvideo/icon.svg"
}
```

**Backward compatibility:** (No breaking changes)
- Existing consumers (`setDefaultConferencingApp`, etc) only use `id`, `title`, `locations`
- New fields are safely ignored by existing code
- No performance impact (same query, indexed fields)

### Frontend - Features Package
- Created `EventTypeListItem.tsx` in `@calcom/features/eventtypes/components/`
- Reusable component for both atoms and web app (same pattern as `ScheduleListItem`)
- Added 11 unit tests

### Frontend - Atoms Package
- `ListEventTypesPlatformWrapper.tsx` - Container component with state management
- `useAtomGetAllEventTypes.ts` - React Query hook for data fetching
- `types.ts` - Type definitions validated against Prisma schema
- Exported `ListEventTypes` from `@calcom/atoms`

### i18n
Added 3 new keys to `apps/web/public/static/locales/en/common.json`:
```json
{
  "cannot_delete_event_type": "This event type cannot be deleted",
  "delete_event_type_confirmation": "Are you sure you want to delete this event type?",
  "error_loading_event_types": "Error loading event types"
}
```

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs).
  - **N/A** - Atoms documentation is maintained separately in Platform docs
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

### Unit Tests
```bash
yarn test packages/features/eventtypes/components/__tests__/EventTypeListItem.test.tsx
```

**Expected:** All 11 tests should pass ✓

Tests cover:
- ✓ Basic rendering (title, description, duration)
- ✓ Conditional rendering (with/without edit URL)
- ✓ Edge cases (null description, long titles)
- ✓ Different durations and locations
- ✓ Dynamic data-testid for E2E

### Manual Testing (in Platform context)

**Prerequisites:**
- Cal.com Platform OAuth client configured
- Valid access token

**Test with CalProvider:**

```tsx
import { CalProvider, ListEventTypes } from "@calcom/atoms";


  <ListEventTypes 
    getEventTypeUrl={(id) => `/event-types/${id}`}
  />

```

**Expected behavior:**
1. Lists all user event types
2. Click "..." shows Edit and Delete options
3. Delete shows browser confirmation
4. Empty state when no event types

### Environment Variables
- **N/A** - Uses Platform authentication context

### Minimal Test Data
- At least 1 event type in the database
- User logged in to Cal.com

### Expected Output (Happy Path)
```json
// API Response: GET /api/v2/atoms/event-types
{
  "status": "success",
  "data": {
    "eventTypes": [
      {
        "id": 1,
        "title": "30min",
        "slug": "30min",
        "description": "Quick 30 minute meeting",
        "length": 30,
        "locations": null,
        "logo": "/app-store/dailyvideo/icon.svg"
      }
    ]
  }
}
```

## Additional Notes

**Architecture decisions:**

1. Component placed in `@calcom/features` for reusability
2. Types validated against Prisma schema and backend response
3. Dynamic `data-testid` for better E2E testability
4. Follows same patterns as existing `ScheduleListItem` component

pedroccastro committed:
feat: add ListEventTypes atom (#25047)

pedroccastro committed:
feat: add ListEventTypes atom for Platform (#25047)

- Add EventTypeListItem component in @calcom/features
- Add ListEventTypesPlatformWrapper with delete functionality
- Implement useAtomGetAllEventTypes hook
- Update backend to include slug, description, length
- Add 11 unit tests (65% coverage)
- Add 3 i18n keys

No breaking changes. Backward compatible.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=25113) <br/>All committers have signed the CLA.

volnei reviewed (dismissed):
Small comment

pedroccastro committed:
refactor(eventtypes): improve EventTypeListItem readability

- Extract EventTypeContent and EventTypeActions sub-components
- Add formatEventTypeDuration helper (90min → '1h 30m')
- Use Badge for duration display (consistency with EventTypeDescription)
- Use polymorphic Wrapper component (DRY)
- Add comprehensive tests
- Reduce main component size

Addresses @volnei feedback

keithwillcode commented:
/tip 75 @pedroccastro

pedroccastro committed:
refactor: replace window.confirm with ConfirmationDialog

- Replace native window.confirm with ConfirmationDialogContent

- Update test mocks for Dialog components

- Fix HTML nesting (div instead of p for Badge container)

- Follows Cal.com pattern from AvailabilitySettings

Addresses @ThyMinimalDev feedback

volnei commented:
@pedroccastro one last thing.. can you resolve the conflicts on en/common.json?

pedroccastro commented:
@volnei Done! Conflicts resolved. 🚀

pedroccastro committed:
fix: add aria-label to options button for accessibility

pedroccastro committed:
test: update getBulkEventTypes test to include new fields (description, slug, length)

pedroccastro commented:
Updated `getBulkEventTypes.test.ts` to expect the new fields (`description`, `slug`, `length`) that were added to the event type select query. The test mock was comparing exact fields and is now aligned with the implementation.

Ran the test locally before pushing, all passing! ✓

ThyMinimalDev committed:
Merge branch 'main' into feat/list-event-types-atom

pedroccastro committed:
Merge branch 'main' into feat/list-event-types-atom

ThyMinimalDev commented on packages/features/eventtypes/components/EventTypeListItem.tsx:
```diff
@@ -0,0 +1,151 @@
+import type { AtomEventTypeListItem } from "../../../platform/atoms/event-types/types";
+import { formatEventTypeDuration } from "../lib/formatEventTypeDuration";
+
+const EventTypeContent = ({ eventType }: { eventType: AtomEventTypeListItem }) => {
```
should this component be directly within atoms/src folder if not used by web ?

pedroccastro committed:
refactor: move EventTypeListItem to atoms package

pedroccastro commented:
Nice @ThyMinimalDev! Moved EventTypeListItem to packages/platform/atoms/event-types/components/
No more cross-package imports between features and atoms 👍

ThyMinimalDev committed:
Merge branch 'main' into feat/list-event-types-atom

pedroccastro committed:
Merge branch 'main' into feat/list-event-types-atom
